src/fsharp/IlxGen.fs

```diff
@@ -3055,7 +3111,61 @@ and GenUntupledArgExpr cenv cgbuf eenv m argInfos expr sequel =
 // Generate calls (try to detect direct calls)
 //--------------------------------------------------------------------------
 
-and GenApp cenv cgbuf eenv (f, fty, tyargs, curriedArgs, m) sequel =
+and GenWitnessArgFromTraitInfo cenv cgbuf eenv m traitInfo =
+    let g = cenv.g
+    let storage = TryStorageForWitness g eenv traitInfo.TraitKey
+    match storage with 
+    | None ->
+        let witnessExpr =
+           ConstraintSolver.CodegenWitnessesForTraitWitness cenv.tcVal g cenv.amap m traitInfo 
+            |> CommitOperationResult
+        match witnessExpr with
+        | Choice1Of2 _traitInfo ->
+            System.Diagnostics.Debug.Assert(false, "expected storage for witness")
+            //failwith "unexpected non-generation of witness "
+        | Choice2Of2 arg ->
+            let eenv = { eenv with suppressWitnesses = true }
+            GenExpr cenv cgbuf eenv SPSuppress arg Continue
+    | Some storage -> 
+        let ty = GenWitnessTy g traitInfo.TraitKey
+        GenGetStorageAndSequel cenv cgbuf eenv m (ty, GenType cenv.amap m eenv.tyenv ty) storage None
+
+and GenWitnessArgFromWitnessInfo cenv cgbuf eenv m witnessInfo =
+    let g = cenv.g
+    let storage = TryStorageForWitness g eenv witnessInfo
+    match storage with 
+    | None ->
+        System.Diagnostics.Debug.Assert(false, "expected storage for witness")
+        //failwith "unexpected non-generation of witness "
```

Comment:
nit: Remove commented code.

Also, since this should not happen, shouldn't we actually fail?